cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/AbstractClusterOperations.java

```diff
@@ -126,10 +138,63 @@ public ClusterDiffResult diff() {
         });
     }
 
-    public abstract void create(String namespace, String name, Handler<AsyncResult<Void>> handler);
+    protected abstract void create(String namespace, String name, Handler<AsyncResult<Void>> handler);
+
+    public void createByName(String namespace, ConfigMap add)   {
```

Comment:
Its a bit strange to call it `deleteByName` when it is called with `ConfigMap` as parameter. Wouldn't `deleteByConfigMap` be bertter name?

Reply:
This is an interesting point, actually. Each operation (create, update, delete) has a `byName()` overload, because in each case it's just the name of the ConfigMap that's passed to `execute()`. `CompositeOperation.getCluster()` usually fetches the ConfigMap with that name and creates a Cluster from it. This is obviously inefficient, but it makes the reconciliation slightly simpler (because there we deal with names, not actual ConfigMaps, as we do in the watcher).

The 'usually' in the above is important, because it's not always the case: `KafkaConnectClusterOperations.delete` creates a cluster from the deployment, and the delete for the `KafkaClusterOperations` uses the statefulset. I guess we could refactor things so that we had a true `deleteByConfigMap()` which avoided the 2nd lookup.

Reply:
Currently the 2nd lookup of the CM or other resource happens within the lock. I suppose that if we pass in a resource that was obtained outside the lock's scope there's a greater risk of concurrent operation, for example there would be a race between an update watch and a periodic reconciliation. On that reasoning we should keep the 2nd lookup.